pkg/gardenlet/controller/controllerinstallation/care/reconciler.go

```diff
@@ -46,8 +46,13 @@ type Reconciler struct {
 func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	log := logf.FromContext(ctx)
 
+	gardenCtx, cancel := context.WithTimeout(ctx, r.Config.SyncPeriod.Duration)
+	defer cancel()
+	seedCtx, cancel := context.WithTimeout(ctx, r.Config.SyncPeriod.Duration/2)
```

Comment:
Can we use the same strategy in gardenlet's `seed/care` and `shoot/care` controllers as well as resource-manager's `health` and `progressing` controllers?